Packages/com.unity.inputsystem/InputSystem/Plugins/XR/GenericXRDevice.cs

```diff
@@ -77,14 +73,35 @@ public override void MakeCurrent()
         }
     }
 
-    public class XRControllerWithRumble : XRController
+    public class XRControllerWithRumble : XRController, IHaptics
     {
-        public SimpleXRRumble rumble { get; private set; }
+        SimpleXRRumble m_Rumble;
+        public SimpleXRRumble rumble { get { return m_Rumble; } }
```

Comment:
Ah doh, I didn't catch before that this is publicly exposed and not just an implementation internal to the controller (like `DualMotorRumble`). Kinda obsoletes my arguments about struct being adequate.

Thinking about it, though, I think it's better to not expose this but have the interface directly on the device. For haptics APIs that can be shared between devices, there should be interfaces derived from `IHaptics` (like `IDualMotorRumble`).

"Is it paused?" is missing from the `IHaptics` interface itself, I think. Overall, the play/pause/resume API can probably be improved.